### DISCUSSION TITLE
Proxy Injection Failure After Cluster Restart

### DISCUSSION THREAD

**Discussion by MattMHB** [linkerd/linkerd2#11113] [General]
Hi, We have a PreProd environment which gets turned off in the evening and back on in the morning. I've noticed that not all the pods get injected so the cluster is only half meshed.

If I restart all the deployments too quickly, they also don't all get injected. I have to restart the deployments one at a time to ensure they are meshed.

Do I have something set up incorrectly? Are there any settings to ensure the proxy injection service is up and running before new pods are created and injected?

I'm running AKS. Thanks

**Comment by MattMHB** [comment]
Has anyone else had the same issue?

**Comment by alpeb** [comment]
Hi @MattMHB you can try setting [webhookFailurePolicy](https://github.com/linkerd/linkerd2/blob/main/charts/linkerd-control-plane/values.yaml#L284-L285) to `Fail`, so that pods that attempt to start before the injector is ready, get rejected. The scheduler should attempt to create them again, until the injector is ready to process them.
